core/src/main/java/org/axonframework/commandhandling/model/inspection/AggregateMemberAnnotatedChildEntityDefinition.java

```diff
@@ -36,22 +36,32 @@
     @Override
     @SuppressWarnings("unchecked")
     public <T> Optional<ChildEntity<T>> createChildDefinition(Field field, EntityModel<T> declaringEntity) {
-        Map<String, Object> attributes = AnnotationUtils.findAnnotationAttributes(field, AggregateMember.class).orElse(null);
+        Map<String, Object> attributes = findAnnotationAttributes(field, AggregateMember.class).orElse(null);
         if (attributes == null
                 || Iterable.class.isAssignableFrom(field.getType())
                 || Map.class.isAssignableFrom(field.getType())) {
             return Optional.empty();
         }
 
         EntityModel entityModel = declaringEntity.modelOf(field.getType());
-        return Optional.of(new AnnotatedChildEntity<>(entityModel,
-                                                      (Boolean) attributes.get("forwardCommands"),
-                                                      (Boolean) attributes.get("forwardEvents"),
-                                                      (msg, parent) -> ReflectionUtils.getFieldValue(field, parent),
-                                                      (msg, parent) -> {
-                                                          Object fieldVal = ReflectionUtils.getFieldValue(field, parent);
-                                                          return fieldVal == null ? emptyList() : singleton(fieldVal);
-                                                      }));
+
+        Boolean forwardEvents = (Boolean) attributes.get("forwardEvents");
+        ForwardingMode eventForwardingMode = (ForwardingMode) attributes.get("eventForwardingMode");
+
+        return Optional.of(new AnnotatedChildEntity<>(
+                entityModel,
+                (Boolean) attributes.get("forwardCommands"),
+                eventForwardingMode(forwardEvents, eventForwardingMode),
+                (String) attributes.get("eventRoutingKey"),
+                (msg, parent) -> ReflectionUtils.getFieldValue(field, parent),
+                (msg, parent) -> {
+                    Object fieldVal = ReflectionUtils.getFieldValue(field, parent);
+                    return fieldVal == null ? emptyList() : singleton(fieldVal);
+                }
+        ));
     }
 
+    private ForwardingMode eventForwardingMode(Boolean forwardEvents, ForwardingMode eventForwardingMode) {
```

Comment:
Instead of "hoping" subclasses invoke this method, provide an implementation of the createChildDefinition that delegates certain call to abstract method. In that case, subclasses cal clearly see which methods they are expected to implement.

Reply:
Fair point, will make this adjustment and use it for all three implementations rather than only for the map/collection implementations.